### PR TITLE
Use setup-python instead of setup-micromamba for pre-commit CI

### PR DESCRIPTION
**Pull request by RMeli**
----
📚 Documentation preview 📚: https://mdanalysisuserguide--388.org.readthedocs.build/en/388/

### CODE REVIEW

**Comment by RMeli** [comment]
We had a failure in #387. Trying to see if using `setup-python` works.

Others encountered the same issue https://github.com/pre-commit/action/issues/214, https://github.com/pre-commit/action/issues/213, https://github.com/pre-commit/action/issues/210. Honestly, the replies from the maintainer on these issues makes me want to steer clear of the `pre-commit` GitHub action.

Additionally:
> this action is in maintenance-only mode and will not be accepting new features.
> generally you want to use [pre-commit.ci](https://pre-commit.ci/) which is faster and has more features.

Maybe we should consider moving to `pre-commit.ci`? @jandom?

File: .github/workflows/pre-commit.yaml [hunk 1]
```diff
@@ -24,19 +24,9 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-
-    - name: setup micromamba
-      uses: mamba-org/setup-micromamba@v1
+    - uses: actions/setup-python@v3
       with:
-        environment-file: environment.yml
```
Comment (orbeckst) [inline .github/workflows/pre-commit.yaml:31]: I assume that the full UG env is not needed for the pre-commit action and that's why you changed to a simple python env?

**Review by orbeckst — APPROVED** [review]
I know very little about the background here. I am providing a "LGTM" because it looks sensible to use a simple python setup to run pre-commit and because you are of the opinion that it will improve the CI and because it didn't visibly break anything.

If someone with more experience could have a look then that would be good, of course, but I ultimately trust that you know what you're doing.

**Comment by RMeli** [comment]
I could not find much information on _why_ this is happening now. But it seems that the action is only in maintenance mode, and the maintainer is very quick at closing issues. In the ones outlining this problem (linked above, _links were wrong and I just updated them_), replies are of the sort

> you are the third or fourth person to not search before creating an issue for this specific thing. please do better!

or

> skill issue imo. it is in the readme you just didn't follow it or read the output of your own job

So not much information there. The README uses the `setup-python` action to setup Python for pre-commit, so that's what I went for and it seems to work as expected.

File: .github/workflows/pre-commit.yaml (hunk 1 above)
Comment (RMeli) [reply .github/workflows/pre-commit.yaml:31]: Yes. The pre-commits builds an environment with its own tools (with the versions specified in the pre-commit configuration).

Comment (orbeckst) [reply .github/workflows/pre-commit.yaml:31]: All makes sense to me, thanks for explaining.

**Comment by orbeckst** [comment]
The maintainer comments on the pre-commit issues are not particularly welcoming. I don't quite know what would be better for a "maintenance-only" project: no response at all or harsh responses + shutting down issues. 🤷

**Review by IAlibay — APPROVED** [review]
Sounds like the precomit action is doing something reasonably nefarious.. I do agree that moving away from it would be good.

**Pull request merged**